// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.hash-back-end-challenge.individualdiscount";
option java_outer_classname = "IndividualDiscountProto";
option objc_class_prefix = "Ind-Disc";

package individualdiscount;

// The individual discount service definition.
service discount {
  // Sends a discount
  rpc IndividualDiscount (IndividualDiscountRequest) returns (IndividualDiscountReply) {}
}

// The request message containing the user's name.
message IndividualDiscountRequest {
  string productId = 1;
  string userId = 2;
}

// The response message containing the greetings
message IndividualDiscountReply {
  string discount = 1;
}
